IoTDeviceSimulator/web-client/src/DeviceTwin.js

```diff
@@ -146,76 +196,402 @@ export function DeviceTwin(props) {
     }, [props]);
 
     const options = [
-        { value: 'Sine Function', label: 'Sine Function' },
-        { value: 'Constant Function', label: 'Constant Function' },
+        { value: 'Sine', label: 'Sine' },
+        { value: 'Constant', label: 'Constant' },
+        { value: 'Linear', label: 'Linear' },
+        { value: 'Noise', label: 'Gaussian Noise' },
+        { value: 'Triangular', label: 'Triangular' },
+        { value: 'Sawtooth', label: 'Sawtooth' },
+        { value: 'Square', label: 'Square' },
     ]
 
-    const handleCloseSineWave = () => {
-        setSineConfig(false);
+    const removeBehaviour = (i) => {
+        let k = 0;
+        for (let i = 0; i < deviceTwin.signalArray.length; i++) {
+            if (deviceTwin.signalArray[i] !== '') {
+                k += 1;
+            }
+        }
+        if (k === 1 && i === 0) {
+            toaster.negative(`There should be atleast one behaviour`)
+        }
+        else {
+            setTabCount(deviceTwin.signalArray.length - 1);
+            deviceTwin.signalArray.splice(i, 1);
+        }
     }
 
-    const handleCloseConstantWave = () => {
-        setConstantConfig(false);
-    }
+    ar = [];
+    arr = [];

```

Comment:
Is there a need of 3 different arrays to store data points? ar, arr and currDataArray?